### ISSUE TITLE
Need for new API that returns the JDBC properties for a specific database

### ISSUE THREAD

**Issue by sorin-pop** [djavorszky/ddn#225]
Due to LiferayUP issue number 366 (https://github.com/vilmospapp/liferay-support-ee/issues/366), I think we would need a proper CloudDB API point which gets as input an agent name and database name, and returns as output the values of the four Liferay JDBC properties:

jdbc.default.driverClassName=
jdbc.default.url=
jdbc.default.username=
jdbc.default.password=

just as they are displayed now by clicking the I(nformation) icon on the CloudDB UI.

For the LiferayUP issue above, it would even be enough to just return the username and password, at least.

This way the code on the LiferayUP side could be more stable and clear, not something like we have now:

```
if (upConfig.getDBLocation() == DBLocation.CLOUDDB) {
    if ((_props.getValue("clouddb.password")!= null) &&
        !_props.getValue("clouddb.password").equals(
            StringPool.EMPTY) &&
        (dbType != DBType.ORACLE.getName())) {
            userpwd = _props.getValue("clouddb.password");
        }
        else {
            userpwd = dbName.toString();
    }
}
```

which turns out to run into the above issue now.

@djavorszky what do you think? As I remember, I did request something similar before, but you did not agree, on the grounds that "CloudDB is meant to be something more general, not only for using with Liferay" or something like that. 

OK, I understand that, but currently we are **practically using it only from LiferayUP** as a third party software. And anyway, if we would have an API that just returns the JDBC properties needed to connect to an existing CloudDB database, that would still be quite "general" (not only Liferay can connect with JDBC to a database, but any other Java application). And this would hugely help us in LiferayUP and would make the whole code easier and clearer, and more stable in LiferayUP. For example, right now, for this issue, I don't see how I could get the password of an existing Oracle database in CloudDB. I don't see any other solution than having some kind of API for it.

**Comment by djavorszky** [comment]
Well, there _is_ an endpoint for something like that, /portalext/:id, though that needs an ID and not the name of the agent and the name of the database.

I think we could add something like this, though I'd not add the "keys" (e.g. `jdbc.default.*=` part) as that is kind of Liferay specific, only the values.

So for example we could have a new endpoint at (actual location, method and url subject to change):
GET /api/dbaccess?agent=${agentname}&dbname=${dbname}

and it would return a json string such as (json key names subject to change):
```
{
    "driver-classname": "com.mysql..",
    "url": "cloud-db.liferay.int",
    "user": "Test1",
    "password": "password1"
}
```

What do you think?
Dani

**Comment by sorin-pop** [comment]
Yep, that's exactly what I'd need. I will do it if you don't have time/wanna bother.

**Comment by sorin-pop** [comment]
Pull request: https://github.com/djavorszky/ddn/pull/226

**Comment by djavorszky** [comment]
Merged #226, will push to live next year
